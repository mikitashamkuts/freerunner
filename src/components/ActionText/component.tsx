import React, {FC, memo} from 'react';
import {View} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';
import {Text} from '../../components';

import {Props} from '.';

const ActionText: FC<Props> = ({text, containerStyle, ...props}) => {
  return (
    <View style={containerStyle}>
      <Text
        fontWeight="Regular"
        fontSize="Regular"
        color="Action"
        text={text}
        numberOfLines={1}
        {...props}
      />
    </View>
  );
};

ActionText.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(ActionText);
