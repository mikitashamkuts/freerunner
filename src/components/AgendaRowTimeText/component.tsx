import React, {FC, memo} from 'react';
import {View} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';
import {Text} from '../../components';

import {Props} from '.';

/**
 * AgendaRowTimeText component displays a text in a view with specific styles.
 *
 * @param {AgendaRowTimeTextProps} props - The props for the component.
 * @returns {JSX.Element} The AgendaRowTimeText component.
 */
const AgendaRowTimeText: FC<Props> = ({text, containerStyle, ...props}) => {
  return (
    <View style={containerStyle}>
      <Text
        fontWeight="Regular"
        fontSize="Regular"
        color="Faded"
        text={text}
        numberOfLines={1}
        {...props}
      />
    </View>
  );
};

AgendaRowTimeText.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(AgendaRowTimeText);
