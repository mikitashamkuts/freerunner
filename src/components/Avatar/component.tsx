import {CachedImage} from '@georstat/react-native-image-cache';
import React, {FC, memo} from 'react';
import {StyleSheet, View} from 'react-native';

import {Props, styles} from '.';

const smallAvatarSize = 48;
const bigAvatarSize = 72;
const logoToAvatarProportion = 75; // %

const Avatar: FC<Props> = ({uri, size, containerStyle}) => {
  const contentContainerStyles = StyleSheet.create({
    smallContainer: {
      height: smallAvatarSize,
      width: smallAvatarSize,
    },
    smallContentContainer: {
      height: (smallAvatarSize / 100) * logoToAvatarProportion,
      width: (smallAvatarSize / 100) * logoToAvatarProportion,
    },
    bigContainer: {
      height: bigAvatarSize,
      width: bigAvatarSize,
    },
    bigContentContainer: {
      height: (bigAvatarSize / 100) * logoToAvatarProportion,
      width: (bigAvatarSize / 100) * logoToAvatarProportion,
    },
  });

  return (
    <View style={[styles.container, contentContainerStyles[`${size}Container`], containerStyle]}>
      <CachedImage source={uri} style={contentContainerStyles[`${size}ContentContainer`]} />
    </View>
  );
};

export default memo(Avatar);
