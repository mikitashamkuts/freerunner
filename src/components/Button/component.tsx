import React, {FC, memo, useCallback} from 'react';
import {Pressable} from 'react-native';

import {Text} from '@src/components';
import {getFunctionTryCatchWrapped as tryCatch} from '@src/utils';

import {whyDidItRenderConfig} from '../../../debug';

import {Props, styles} from '.';

/**
 * Button component renders a pressable button with text and handles the press event safely.
 *
 * @param {ButtonProps} props - The props for the component.
 * @returns {JSX.Element} The Button component.
 */
const Button: FC<Props> = ({onPress, text, accessibilityLabel, accessibilityHint}) => {
  const handleOnPress = useCallback(() => {
    tryCatch(function handleOnPressSafe() {
      onPress?.();
    })();
  }, [onPress]);

  return (
    <Pressable
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      onPress={handleOnPress}
      style={styles.container}>
      <Text numberOfLines={1} color="Light" text={text} />
    </Pressable>
  );
};

Button.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(Button);
