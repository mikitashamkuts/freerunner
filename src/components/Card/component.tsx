import React, {FC, memo} from 'react';
import {View} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';

import {Props, styles} from '.';

const Card: FC<Props> = ({containerStyle, children}) => {
  return <View style={[styles.container, containerStyle]}>{children}</View>;
};

Card.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(Card);
