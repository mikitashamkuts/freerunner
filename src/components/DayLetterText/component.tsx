import React, {FC, memo, useMemo} from 'react';
import {View, useColorScheme} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';
import {Text} from '../../components';
import {getFunctionTryCatchWrapped as tryCatch} from '../../utils';

import {Props} from '.';

const DayLetterText: FC<Props> = ({text, containerStyle, ...props}) => {
  const isDarkMode = useColorScheme() === 'dark';

  const color = useMemo(() => {
    return tryCatch(function getDayLetterTextColor() {
      return isDarkMode ? 'Light' : 'Dark';
    })();
  }, [isDarkMode]);

  return (
    <View style={containerStyle}>
      <Text
        fontWeight="SemiBold"
        fontSize="Small"
        color={color}
        text={text}
        numberOfLines={1}
        {...props}
      />
    </View>
  );
};

DayLetterText.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(DayLetterText);
