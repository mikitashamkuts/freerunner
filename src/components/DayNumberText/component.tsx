import React, {FC, memo} from 'react';
import {View, useColorScheme} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';
import {Text} from '../../components';

import {Props} from '.';

const DayNumberText: FC<Props> = ({text, containerStyle, ...props}) => {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    <View style={containerStyle}>
      <Text
        fontWeight="Regular"
        fontSize="Regular"
        color={isDarkMode ? 'Light' : 'Dark'}
        text={text}
        numberOfLines={1}
        {...props}
      />
    </View>
  );
};

DayNumberText.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(DayNumberText);
