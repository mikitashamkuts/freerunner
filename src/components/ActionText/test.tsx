/**
 * @format
 */

import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

import {default as ActionText} from './component';

describe('ActionText Component', () => {
  it('Renders with default props and matches previous snapshot: ', () => {
    expect(renderer.create(<ActionText text={'ActionText'} />).toJSON()).toMatchSnapshot();
  });

  it('Renders with custom props and matches previous snapshot: ', () => {
    expect(
      renderer
        .create(
          <ActionText fontWeight="SemiBold" fontSize="Small" color="Faded" text={'ActionText'} />,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
