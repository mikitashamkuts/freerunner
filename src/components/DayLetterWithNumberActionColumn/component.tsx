import React, {FC, memo, useCallback, useMemo} from 'react';
import {Pressable, useColorScheme} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';
import {getFunctionTryCatchWrapped as tryCatch} from '../../utils';
import {DayLetterText} from '../DayLetterText';
import {DayNumberText} from '../DayNumberText';

import {Props, styles} from '.';

const DayLetterWithNumberActionColumn: FC<Props> = ({isActive, dayLetter, dayNumber, onPress}) => {
  const isDarkMode = useColorScheme() === 'dark';

  const handleOnPress = useCallback(() => {
    tryCatch(function handleOnPressSafe() {
      onPress?.();
    })();
  }, [onPress]);

  const dayNumberTextColor = useMemo(() => {
    return tryCatch(function getDayNumberTextColor() {
      return (isActive && 'Light') || isDarkMode ? 'Light' : 'Dark';
    })();
  }, [isActive, isDarkMode]);

  return (
    <Pressable onPress={handleOnPress} style={styles.container}>
      <DayLetterText text={dayLetter} containerStyle={styles.dayLetterTextContainer} />
      <DayNumberText
        text={dayNumber}
        containerStyle={[
          styles.dayNumberTextContainer,
          isActive ? styles.dayNumberTextContainerActive : {},
        ]}
        color={dayNumberTextColor}
      />
    </Pressable>
  );
};

DayLetterWithNumberActionColumn.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(DayLetterWithNumberActionColumn);
