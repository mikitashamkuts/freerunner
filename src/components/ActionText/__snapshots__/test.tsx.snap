// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionText Component Renders with custom props and matches previous snapshot:  1`] = `
<View>
  <Text
    accessibilityRole="text"
    numberOfLines={1}
    style={
      [
        {
          "color": "#8E8E93",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 10,
        },
        undefined,
      ]
    }
  >
    ActionText
  </Text>
</View>
`;

exports[`ActionText Component Renders with default props and matches previous snapshot:  1`] = `
<View>
  <Text
    accessibilityRole="text"
    numberOfLines={1}
    style={
      [
        {
          "color": "#FF382B",
          "fontFamily": "Inter-Regular",
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    ActionText
  </Text>
</View>
`;
